src/main/java/com/spotify/confidence/Confidence.java

```diff
@@ -214,12 +214,8 @@ public Confidence build() {
       final FlagResolverClient flagResolverClient =
           new FlagResolverClientImpl(
               new GrpcFlagResolver(clientSecret, flagResolverManagedChannel));
-      final SystemClock clock = new SystemClock();
-      final GrpcEventUploader uploader =
-          new GrpcEventUploader(clientSecret, clock, DEFAULT_CHANNEL);
```

Comment:
Who owns this channel? It's currently being passed to `GrpcEventUploader` which then closes it in the close method. IMO if passed in it's not owned by `GrpcEventUploader` and the owner should close it